### PR TITLE
Get methods

### PR DESCRIPTION
**Pull request by dehsani-academic**
get methods were updated to conform to matrix tables
tests were fixed

### CODE REVIEW

**Comment by physikerwelt** [comment]
I think you need to pin the dependencies in setup.cfg With the latest version of the requirements there is an error.

> E   ImportError: cannot import name '_endpoint_from_view_func' from 'flask.helpers' (/usr/local/lib/python3.7/site-packages/flask/helpers.py)

**Comment by dehsani-academic** [comment]
> I think you need to pin the dependencies in setup.cfg With the latest version of the requirements there is an error.
> 
> > E   ImportError: cannot import name '_endpoint_from_view_func' from 'flask.helpers' (/usr/local/lib/python3.7/site-packages/flask/helpers.py)

setting Flask version to 1.1.2 resolves that issue.  
database migrations now need to be updated.

**Comment by fmux** [comment]
> setting Flask version to 1.1.2 resolves that issue.
I think it's generally a bad idea to pin the version down to the release level. This way you will have to watch for security patches and bump the version yourself every time. In the present case, I think it would be entirely enough to pin `Flask>=1,<2`.

Note however that the issue seems to have been fixed in flask-restful 0.3.9 anyway (cf. https://github.com/flask-restful/flask-restful/pull/915).

**Comment by physikerwelt** [comment]
> > I think it's generally a bad idea to pin the version down to the release level. This way you will have to watch for security patches and bump the version yourself every time. In the present case, I think it would be entirely enough to pin `Flask>=1,<2`.

@dehsani-academic can you try this? I think it would be best to do this in a separate pull request.

**Comment by dehsani-academic** [comment]
> > > I think it's generally a bad idea to pin the version down to the release level. This way you will have to watch for security patches and bump the version yourself every time. In the present case, I think it would be entirely enough to pin `Flask>=1,<2`.
> 
> @dehsani-academic can you try this? I think it would be best to do this in a separate pull request.

I saw your comment too late; had already added the range for version in latest commit.  But it works.

**Review by physikerwelt — COMMENTED** [review]
Does this change imply that  the database is be psql? I will update the Testserver with this version and get back to this pull request.

File: startup.sh [hunk 1]
```diff
@@ -1,6 +1,9 @@
 #!/bin/bash
 flask manage_db drop_all
```
Comment (physikerwelt) [inline startup.sh:2]: it is a bit couter-intutive that this script drops all databases no matter what.

**Review by physikerwelt — APPROVED** [review]
This pull request is not atomic. I code review, without testing looks good. I will merge it as it is, but please try to make smaller atomic pull requests in the future.

**Comment by comArkGh8** [comment]
> Does this change imply that the database is be psql? I will update the Testserver with this version and get back to this pull request.

Yes, it should be postgresql.

**Comment by dehsani-academic** [comment]
> Does this change imply that the database is be psql? I will update the Testserver with this version and get back to this pull request.

yes, it should be a postgresql database.

File: startup.sh (hunk 1 above)
Comment (dehsani-academic) [reply startup.sh:2]: how is it counterintuitive?

Comment (physikerwelt) [reply startup.sh:2]: you run startup.sh and all your data will be deleted. A common practice is to be extra careful before permanently deleting data. See for example https://uxmovement.com/buttons/how-to-make-sure-users-dont-accidentally-delete/

**Pull request merged**